src/services/housing.js

```diff
@@ -0,0 +1,202 @@
+ * @param {String} editorUsername the student username of the person filling out the application
+ * @param {ApartmentApplicant[]} applicants Array of ApartmentApplicant objects
+ * @param {ApartmentChoice[]} apartmentChoices Array of ApartmentChoice objects
+ * @return {Promise.<Number>} Application's ID number
+ */
+const saveApartmentApplication = async (
+  applicationID,
+  editorUsername,
+  applicants,
+  apartmentChoices,
+) => {
+  let applicationDetails = {
+    ApplicationID: applicationID ?? -1,
+    EditorUsername: editorUsername,
+    Applicants: applicants.map((applicant) => ({
+      ApplicationID: applicant?.ApplicationID ?? applicationID ?? -1,
+      Username: applicant.Profile.AD_Username,
+      OffCampusProgram: applicant?.OffCampusProgram ?? '',
+    })),
+    ApartmentChoices: apartmentChoices.map((apartmentChoice) => ({
+      ApplicationID: apartmentChoice?.ApplicationID ?? applicationID ?? -1,
+      ...apartmentChoice,
+    })),
+  };
+  if (applicationID) {
+    return await http.put(`housing/apartment/applications/${applicationID}/`, applicationDetails);
+  } else {
+    return await http.post(`housing/apartment/applications/`, applicationDetails);
+  }
+};
+
+/**
+ * Update the application editor of the application to the database
+ * @param {Number} applicationID the application ID number
+ * @param {String} newEditorUsername the student username of the person who will be allowed to edit this application
+ * @return {Promise.<Boolean>} Status of whether or not the operation was successful
+ */
+const changeApartmentAppEditor = async (applicationID, newEditorUsername) => {
+  let newEditorDetails = {
+    ApplicationID: applicationID,
+    EditorUsername: newEditorUsername,
+  };
+  return await http.put(
+    `housing/apartment/applications/${applicationID}/editor/`,
+    newEditorDetails,
+  );
+};
+
+/**
+ * Get active apartment application for given application ID number
+ * @param {Number} applicationID the application ID number for the desired application
+ * @return {Promise.<ApplicationDetails>} Application details
+ */
+const getApartmentApplication = async (applicationID) => {
+  return await http.get(`housing/apartment/applications/${applicationID}/`);
+};
+
+/**
+ * Get active apartment applications for the current semester
+ * @return {Promise.<ApplicationDetails>[]} Application details
+ */
+const getAllApartmentApplications = async () => {
+  let applicationDetailsArray = await http.get(`housing/admin/apartment/applications/`);
+
+  // Calculate the total and average points for each application
+  applicationDetailsArray.forEach((applicationDetails) => {
+    let totalPoints = 0;
+    applicationDetails.Applicants.forEach((applicant) => {
+      totalPoints += applicant.Points;
+    });
+    let avgPoints = totalPoints / applicationDetails.Applicants.length;
+    applicationDetails.TotalPoints = totalPoints;
+    applicationDetails.AvgPoints = avgPoints;
+  });
```

Comment:
Could this not be done in the API? It seems odd to fetch data from the API and then immediately format that data again. Usually, an API endpoint should do all the general puprose data transformation ahead of time, so the  front end requests the data in it's final form and is only responsible for displaying it.

Reply:
Thank you for pointing this out, I forgot about this.

This was discussed back and forth quite a lot, and yes it could easily be done in the API. I think before we implemented it, there were reasons that we thought it would be better to put it here, but the process of implementing that endpoint revealed various things about how best to handle that stuff, and so I believe the actual implementation of the endpoint now would make that calculation super easy to put in the backend. 
I think the reasoning before had to do with the points being stored with the applicant data in the AA_Applicants table in the database, and thus the applicants, apartment choices, and such would have been retrieved somewhat separately, but we instead figured out that it is more efficient for both UI and API to handle it all in the current nested JSON structure, so the points are now generated/calculated at the time that the data is loaded or retrieved for the staff menu. 
TLDR our original design _we thought_ that calculating the point in the frontend made more sense, but in the new/current design it is obvious that the points total and average should be in the API, and I will make this change tonight

Reply:
This is now ready in #1066